cmd/snap/cmd_buy.go

```diff
@@ -0,0 +1,132 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"strings"
+
+	"github.com/snapcore/snapd/client"
+	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/store"
+
+	"github.com/jessevdk/go-flags"
+)
+
+var shortBuyHelp = i18n.G("Finds packages to install")
+var longBuyHelp = i18n.G(`
+The find command queries the store for available packages.
+`)
+
+var positiveResponse = map[string]bool{
+	"":            true,
+	i18n.G("y"):   true,
+	i18n.G("yes"): true,
+}
+
+type cmdBuy struct {
+	Currency  string `long:"currency" description:"ISO 4217 code for currency (https://en.wikipedia.org/wiki/ISO_4217)"`
+	Channel   string `long:"channel" description:"Use this channel instead of stable"`
+	BackendID string `long:"backend-id" description:"e.g. \"credit_card\", \"rest_paypal\""`
+	MethodID  int    `long:"method-id" description:"numeric identifier for a specific credit card or Paypal account"`
+
+	Positional struct {
+		SnapName string `positional-arg-name:"<snap-name>"`
+	} `positional-args:"yes" required:"yes"`
+}
+
+func init() {
+	addCommand("buy", shortBuyHelp, longBuyHelp, func() flags.Commander {
+		return &cmdBuy{}
+	})
+}
+
+func (x *cmdBuy) Execute([]string) error {
+	return buySnap(&store.BuyOptions{
+		SnapName:  x.Positional.SnapName,
+		Currency:  x.Currency,
+		BackendID: x.BackendID,
+		MethodID:  x.MethodID,
+		Channel:   x.Channel,
+	})
+}
+
+func buySnap(opts *store.BuyOptions) error {
+	cli := Client()
+
+	snaps, resultInfo, err := cli.Find(&client.FindOptions{
+		Query: fmt.Sprintf("name:%s", opts.SnapName),
+	})
+
+	if err != nil {
+		return err
+	}
+
+	if len(snaps) < 1 {
+		return fmt.Errorf(i18n.G("cannot buy snap %q: it cannot be found"), opts.SnapName)
+	}
```

Comment:
if `len(snaps)>1` then something funny is going on and you should bail.

You probably also want to check `Snapname` doesn't have any of `:*` to provide  a nicer error message